compatibility/bazel_tools/testing.bzl

```diff
@@ -6,277 +6,174 @@ load(
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.1.0-snapshot.20200506.4107.0.7e448d81": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.1.1": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200513.4172.0.021f4af3": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200520.4224.0.2af134ca": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200520.4228.0.595f1e27": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "0.0.0": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
 }
 
 def get_excluded_tests(test_tool_version, sandbox_version):
-    return excluded_test_tool_tests.get(test_tool_version, default = {}).get(sandbox_version, default = [])
+    exclusion_ranges = excluded_test_tool_tests.get(test_tool_version, default = [])
+    exclusions = []
+    for range in exclusion_ranges:
+        start = range.get("start")
+        end = range.get("end")
+        if start and sandbox_version != "0.0.0" and cmp_version(sandbox_version, start) < 0:
+            continue
+        if end and (cmp_version(sandbox_version, end) > 0 or sandbox_version == "0.0.0"):
+            continue
```

Comment:
Skylib has a reasonably nice [API for comparing versions](https://github.com/bazelbuild/bazel-skylib/blob/560d7b2359aecb066d81041cb532b82d7354561b/lib/versions.bzl#L57-L80) I think it might make this a bit easier to read. Something along the lines of

```
if versions.is_at_least(start, sandbox_version) and versions.is_at_most(end, sandbox_version):
    exclusions += range["exclusions"]
```

Assuming that our `is_at_least` and `is_at_most` treats `0.0.0` as most recent.

Reply:
I’ve looked at this API but it only uses `major,minor,patch` and ignores the prerelease part.

Reply:
Yes, the suggestion is to replicate that API, not to use it.

Reply:
Oh that makes sense, I’ll give it a shot :+1:

Reply:
I’ve put the skylib-style API on top of `cmp_versions` and made that one private.

Reply:
Thank you, looks good!